build.gradle

```diff
@@ -155,3 +152,22 @@ task printVersion {
         }
     }
 }
+
+// run each test for every JAVA_HOME_* environment which is specified and set as environment
+[sourceTarget, jvmToTest.split(",")].combinations { targetJVM, envJVM ->

```

Comment:
Here, sourceTarget will be used as character array, so it will create problems as soon as you switch to sourceTarget="10". This should probably read:
```
[[sourceTarget], jvmToTest.split(",")]...
```